### DISCUSSION TITLE
Imrpoved debugging output for slow indexing

### DISCUSSION THREAD

**Discussion by alita-moore** [microsoft/pylance-release#6177] [Enhancement]
Currently when I start vscode for the first time and it indexes my workspace, it tends to take a long time and it outputs the following error:
```
2024-07-12 15:37:56.583 [info] [Error - 3:37:56 PM] (3199) Enumeration of workspace source files is taking longer than 10 seconds.
This may be because:
* You have opened your home directory or entire hard drive as a workspace
* Your workspace contains a very large number of directories and files
* Your workspace contains a symlink to a directory with many files
* Your workspace is remote, and file enumeration is slow
To reduce this time, open a workspace directory with fewer files or add a pyrightconfig.json configuration file with an "exclude" section to exclude subdirectories from your workspace. For more details, refer to https://github.com/microsoft/pyright/blob/main/docs/configuration.md.
```
However, it would be helpful to have more insights about which files and directories are taking longer than others so that I can take action more effectively. For example, having a list of directories and the time it took for pylance to index them.

**Comment by StellaHuang95** [comment]
Thanks for reaching out. The error you're encountering happens during the enumerations of files in your workspace, not during the indexing step. The process is taking longer than 10 seconds as indicated by the log. Could you please check the root of your workspace to see if it matches any of the scenarios described in the error message? Addressing these factors should help speed up the file enumeration process.

**Comment by alita-moore** [comment]
I just mean it would be nice if it gave a more clear output in terms of which folders were taking a long time. For example, if I had accidentally included the .venv directory it would be nice to see that it's taking a long time so I can exclude from the pyright config.

**Comment by StellaHuang95** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
